contrib/terraform/openstack/modules/compute/main.tf

```diff
@@ -19,12 +19,13 @@ resource "openstack_networking_secgroup_rule_v2" "k8s_master" {
 }
 
 resource "openstack_networking_secgroup_v2" "bastion" {
-  name        = "${var.cluster_name}-bastion"
+  count       = "${var.number_of_bastions ? 1 : 0}"
+  name        = "${format("%s-bastion", var.cluster_name)}"
   description = "${var.cluster_name} - Bastion Server"
 }
 
 resource "openstack_networking_secgroup_rule_v2" "bastion" {
-  count             = "${length(var.bastion_allowed_remote_ips)}"
+  count             = "${var.number_of_bastions ? length(var.bastion_allowed_remote_ips) : 0}"
```

Comment:
If `var.number_of_bastions`, isn't `length(var.bastion_allowed_remote_ips)` already 0 ?

Reply:
agreed. fixed/

Reply:
OK. got it.